src/Elmish.WPF/Merge.fs

```diff
@@ -15,22 +15,58 @@ type DuplicateIdException (sourceOrTarget: SourceOrTarget, index1: int, index2:
   member this.Index2 = index2
   member this.Id = id
 
+type internal CollectionTarget<'a> =
+  { GetLength: unit -> int
+    GetAt: int -> 'a
+    Append: 'a -> unit
+    InsertAt: int * 'a -> unit
+    ReplaceAt: int * 'a -> unit
```

Comment:
Let's call this `SetAt`

Reply:
F# Core has [List.updateAt](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-collections-listmodule.html#updateAt), if that matters. It also has `List.item` where we use `GetAt`, though that's potentially less clear.

Reply:
@the-not-mad-psychologist The problem with `updateAt` is that we already have that term overloaded in this project (MVU where the U is a giant `Update`). And this particular function is actually *not* that update so I think we're good using `SetAt` here.